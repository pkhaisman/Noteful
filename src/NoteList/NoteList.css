.NoteList {
    flex-grow: 2;
    list-style-type: none;
}