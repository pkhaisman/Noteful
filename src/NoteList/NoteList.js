import React from 'react';
import { Link } from 'react-router-dom';
import './NoteList.css';
import Note from '../Note/Note';

// Refactor needed
class NoteList extends React.Component {
    render() {
        // if a folder is clicked this shows the notes within
        if (this.props.folderId) {
            const filteredNotes = this.props.notes.filter(note => note.folderId === this.props.folderId);
            const notes = filteredNotes.map(note => {
                return (
                    <li key={note.id}>
                        <Link to={`note/${note.id}`}>
                            <Note note={note} />
                        </Link>
                    </li>
                )
            });
            return (
                <ul className='NoteList'>
                    {notes}
                </ul>
            );
        // if a note is clicked this shows the singular note
        } else if (this.props.noteId) {
            const note = this.props.notes.find(note => note.id === this.props.noteId)
            return (
                <ul className='NoteList'>
                    <li key={note.id}>
                        <Note 
                            note={note}
                            noteId={this.props.noteId} />
                    </li>
                </ul>
            );
        // this runs when no folder or note is selected. shows all notes
        } else {
            const notes = this.props.notes.map(note => {
                return (
                    <li key={note.id}>
                        <Link to={`/note/${note.id}`}>
                            <Note note={note} />
                        </Link>
                    </li>
                )
            });
            return (
                <ul className='NoteList'>
                    {notes}
                </ul>
            );
        }
    }
}

export default NoteList;