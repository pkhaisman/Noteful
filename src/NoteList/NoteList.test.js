import React from 'react';
import NoteList from './NoteList';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('NoteList Component', () => {

    // does this qualify as a snapshot test?
    it('renders without crashing', () => {
        shallow(<NoteList />)
    });
    
    it('renders the UI as expected', () => {
        const wrapper = shallow(<NoteList />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});