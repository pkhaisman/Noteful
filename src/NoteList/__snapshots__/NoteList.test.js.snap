// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList Component renders the UI as expected 1`] = `<AllNotes />`;
